homeassistant/components/recorder/statistics.py

```diff
@@ -1248,48 +1265,40 @@ def _get_oldest_sum_statistic_in_sub_period(
             else:
                 period = start_time.replace(minute=0, second=0, microsecond=0)
             prev_period = period - table.duration
-            stmt += lambda q: q.filter(table.start == prev_period)
+            stmt += lambda q: q.filter(table.start >= prev_period)
         stats = execute_stmt_lambda_element(session, stmt)
-        return (
-            (stats[0].sum, process_timestamp(stats[0].start)) if stats else (None, None)
-        )
+        return stats[0].sum if stats else None
 
-    oldest_start: datetime | None
     oldest_sum: float | None = None
 
+    if not tail_only and oldest_stat is not None:
+        assert main_start_time is not None
```

Comment:
According to the method signature, it can clearly be `None`, and nothing else is guarding it. It either can be `None` (and should be handled) or the function signature is incorrect. IMHO the `assert` is incorrect in any case.

Reply:
It can't be `None` if `tail_only` is `False`, but I'm not sure how to convey that to mypy. Fixed by adding a check to the if statement.